aws/components/queuehost/setup.ftl

```diff
@@ -0,0 +1,218 @@
+    [@addDefaultGenerationContract subsets=[ "template", "epilogue" ] /]
+[/#macro]
+
+[#macro aws_queuehost_cf_deployment occurrence ]
+    [@debug message="Entering" context=occurrence enabled=false /]
+
+    [#-- Component State helpers --]
+    [#local core = occurrence.Core ]
+    [#local solution = occurrence.Configuration.Solution ]
+    [#local resources = occurrence.State.Resources ]
+
+    [#-- Baseline component lookup --]
+    [#local baselineLinks = getBaselineLinks(occurrence, [ "Encryption" ] )]
+    [#local baselineComponentIds = getBaselineComponentIds(baselineLinks)]
+    [#local cmkKeyId = baselineComponentIds["Encryption"]!"" ]
+
+    [#-- Network Lookup --]
+    [#local networkLink = getOccurrenceNetwork(occurrence).Link!{} ]
+    [#local networkLinkTarget = getLinkTarget(occurrence, networkLink ) ]
+    [#if ! networkLinkTarget?has_content ]
+        [@fatal message="Network could not be found" context=networkLink /]
+        [#return]
+    [/#if]
+    [#local networkConfiguration = networkLinkTarget.Configuration.Solution]
+    [#local networkResources = networkLinkTarget.State.Resources ]
+    [#local vpcId = networkResources["vpc"].Id ]
+
+    [#-- Resources and base configuration --]
+    [#local brokerId = resources["broker"].Id ]
+    [#local brokerFullName = resources["broker"].Name ]
+    [#local brokerShortName = resources["broker"].ShortName ]
+    [#local brokerPorts = resources["broker"].Ports]
+    [#local securityGroupId = resources["sg"].Id ]
+    [#local securityGroupName = resources["sg"].Name ]
+
+    [#local engine = solution.Engine]
+    [#local engineVersion = solution.EngineVersion ]
+
+    [#local networkProfile = getNetworkProfile(solution.Profiles.Network)]
+
+    [#local hibernate = solution.Hibernate.Enabled && isOccurrenceDeployed(occurrence)]
+
+    [#-- Link Processing --]
+    [#list solution.Links?values as link]
+        [#if link?is_hash]
+            [#local linkTarget = getLinkTarget(occurrence, link) ]
+
+            [@debug message="Link Target" context=linkTarget enabled=false /]
+
+            [#if !linkTarget?has_content]
+                [#continue]
+            [/#if]
+
+            [#local linkTargetCore = linkTarget.Core ]
+            [#local linkTargetConfiguration = linkTarget.Configuration ]
+            [#local linkTargetResources = linkTarget.State.Resources ]
+            [#local linkTargetAttributes = linkTarget.State.Attributes ]
+
+            [#if deploymentSubsetRequired(QUEUEHOST_COMPONENT_TYPE, true)]
+                [@createSecurityGroupRulesFromLink
+                    occurrence=occurrence
+                    groupId=cacheSecurityGroupId
+                    linkTarget=linkTarget
+                    inboundPorts=[ port ]
+                /]
+            [/#if]
+
+        [/#if]
+    [/#list]
+
+    [#-- Secret Management --]
+    [#local secretStoreLink = getLinkTarget(occurrence, solution.RootCredentials.SecretStore) ]
```

Comment:
Need to check the result to ensure link resolved correctly

Reply:
Fixed